### PR TITLE
Issue/183

### PR DESCRIPTION
**Pull request by hayleyavw**
Alrighty, here it is, the PR you have all been waiting for. Grab a drink and some snacks, mark yourself as "busy" on Slack, and get comfy 😎 

This PR fixes:
- #188 and #114 Restructured the loaders and created more helpful error messages for authors
- #183 All binary numbers content that has been marked as complete on the scorecard (excluding resources) as well as one programming exercise just to test things/have a programming exercise to use for playing with layout/css later.
- #182 Curriculum areas have been added.
- #179 All the placeholder content is gone and we can now start playing with real content!

This PR also adds notes to the docs about the loaders, and updated the docs about adding content to reflect the restructuring.

Issues that will need to be created after merging with develop:
- Review added content
- Figure out suitable layout and styling for lessons (we really need to add css for panels)
- Figure out what resources need to be created for binary numbers topic

### CODE REVIEW

File: csunplugged/topics/management/commands/_ProgrammingExercisesLoader.py [hunk 1]
```diff
@@ -2,35 +2,177 @@
...
+        self.BASE_PATH = os.path.join(self.BASE_PATH, os.path.split(structure_file_path)[0])
         self.topic = topic
 
     def load(self):
-        """Load the content for programming exercises."""
-        programming_exercises_structure = self.load_yaml_file(self.structure_file)
+        """Load the content for programming exercises.
+
+        Raises:
+            MissingRequiredFieldError:
```
Comment (JackMorganNZ) [inline csunplugged/topics/management/commands/_ProgrammingExercisesLoader.py]: Are the `:` at the end of these lines signifying there is a description missing after it?

File: csunplugged/resources/views/sorting_network.py [hunk 2]
```diff
@@ -14,7 +14,7 @@ def resource_image(get_request, resource):
     Returns:
         A Pillow image object.
     """
-    image_path = 'static/img/resource-sorting-network-colour.png'
+    image_path = 'static/img/resources/resource-sorting-network-colour.png'
```
Comment (JackMorganNZ) [inline csunplugged/resources/views/sorting_network.py]: Need to establish rule on which quotes to use.

File: csunplugged/utils/BaseLoader.py [hunk 3]
```diff
@@ -36,29 +44,48 @@ def setup_md_to_html_converter(self):
...
         custom_processors = self.converter.processor_defaults()
-        custom_processors.add('remove-title')
+        custom_processors.add("video")
+        custom_processors.add("remove-title")
         self.converter.update_processors(custom_processors)
 
-    def convert_md_file(self, md_file_path):
-        """Convert the given Markdown file to HTML.
+    def convert_md_file(self, md_file_path, config_file_path, heading_required=True):
+        """Return sthe Verto object for a given Markdown file.
```
Comment (JackMorganNZ) [inline csunplugged/utils/BaseLoader.py]: 's' need to be removed before 'the'

File: csunplugged/utils/BaseLoader.py [hunk 4]
```diff
@@ -36,29 +44,48 @@ def setup_md_to_html_converter(self):
...
+            "markdown.extensions.fenced_code",
+            "markdown.extensions.codehilite",
+            "markdown.extensions.sane_lists",
+            "markdown.extensions.tables",
             mdx_math.MathExtension(enable_dollar_delimiter=True)
         ]
         self.converter = Verto(html_templates=templates, extensions=extensions)
         custom_processors = self.converter.processor_defaults()
-        custom_processors.add('remove-title')
+        custom_processors.add("video")
```
Comment (JackMorganNZ) [inline csunplugged/utils/BaseLoader.py]: I believe this is a pointless statement, as `video` should be enabled by default.

File: csunplugged/utils/BaseLoader.py [hunk 5]
```diff
@@ -36,29 +44,48 @@ def setup_md_to_html_converter(self):
...
         self.converter.update_processors(custom_processors)
 
-    def convert_md_file(self, md_file_path):
-        """Convert the given Markdown file to HTML.
+    def convert_md_file(self, md_file_path, config_file_path, heading_required=True):
+        """Return sthe Verto object for a given Markdown file.
 
         Args:
-            md_file_path: location of the markdown file to convert
+            md_file_path: location of md file to convert
```
Comment (JackMorganNZ) [inline csunplugged/utils/BaseLoader.py]: Could we use the full English, rather than abbreviations in docstrings? Makes them much easier to read.

File: csunplugged/utils/BaseLoader.py [hunk 6]
```diff
@@ -79,31 +106,55 @@ def load_yaml_file(self, yaml_file_path):
         Args:
             file_path: location of yaml file to read
 
-        Returns:
-             Either list or string, depending on structure of given yaml file
+        Return:
```
Comment (JackMorganNZ) [inline csunplugged/utils/BaseLoader.py]: Should be `Returns:`

File: csunplugged/utils/BaseLoader.py [hunk 7]
```diff
@@ -36,29 +44,48 @@ def setup_md_to_html_converter(self):
...
+    def convert_md_file(self, md_file_path, config_file_path, heading_required=True):
+        """Return sthe Verto object for a given Markdown file.
 
         Args:
-            md_file_path: location of the markdown file to convert
+            md_file_path: location of md file to convert
 
-        Returns:
-            Verto result object
+        Return:
```
Comment (JackMorganNZ) [inline csunplugged/utils/BaseLoader.py]: Should be `Returns:`

File: csunplugged/utils/BaseLoader.py [hunk 8]
```diff
@@ -79,31 +106,55 @@ def load_yaml_file(self, yaml_file_path):
...
+        if isinstance(yaml_contents, dict) is False:
+            raise InvalidConfigFileError(yaml_file_path)
+
+        return yaml_contents
 
     def load_template_files(self):
         """Load custom HTML templates for converter.
 
-        Returns:
+        Return:
```
Comment (JackMorganNZ) [inline csunplugged/utils/BaseLoader.py]: Should be `Returns:` and 4 spaces indentation on next line (not 3).

File: csunplugged/utils/errors/CouldNotFindConfigFileError.py [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+    def __init__(self, config_file_path):
+        """Create the error for missing config file."""
+        super().__init__()
+        self.config_file_path = config_file_path
+
+    def __str__(self):
+        """
+        Override default error string.
+
+        Return:
```
Comment (JackMorganNZ) [inline csunplugged/utils/errors/CouldNotFindConfigFileError.py]: Should be `Returns:`

File: csunplugged/utils/errors/CouldNotFindConfigFileError.py [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+          Error message for missing config file.
```
Comment (JackMorganNZ) [inline csunplugged/utils/errors/CouldNotFindConfigFileError.py]: Line should be indented by 4 spaces, instead of 2.

File: csunplugged/utils/errors/CouldNotFindImageError.py [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+        """Create the error missing image."""
+        super().__init__()
+        self.image_path = image_path
+        self.reference_file_path = reference_file_path
+
+    def __str__(self):
+        """
+        Override default error string.
+
+        Return:
```
Comment (JackMorganNZ) [inline csunplugged/utils/errors/CouldNotFindImageError.py]: Should be `Returns:` and 4 spaces indentation on next line.

File: csunplugged/utils/errors/CouldNotFindMarkdownFileError.py [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+        """Create the error for missing markdown file."""
+        super().__init__()
+        self.md_file_path = md_file_path
+        self.config_file_path = config_file_path
+
+    def __str__(self):
+        """
+        Override default error string.
+
+        Return:
```
Comment (JackMorganNZ) [inline csunplugged/utils/errors/CouldNotFindMarkdownFileError.py]: Should be `Returns:` and 4 spaces indentation on next line.

File: csunplugged/utils/errors/EmptyConfigFileError.py [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+    def __init__(self, yaml_file_path):
+        """Create the error for empty config file."""
+        super().__init__()
+        self.yaml_file_path = yaml_file_path
+
+    def __str__(self):
+        """
+        Override default error string.
+
+        Return:
```
Comment (JackMorganNZ) [inline csunplugged/utils/errors/EmptyConfigFileError.py]: Should be `Returns:` and 4 spaces indentation on next line.

File: csunplugged/utils/errors/EmptyMarkdownFileError.py [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+    def __init__(self, md_file_path):
+        """Create the error for empty markdown file."""
+        super().__init__()
+        self.md_file_path = md_file_path
+
+    def __str__(self):
+        """
+        Override default error string.
+
+        Return:
```
Comment (JackMorganNZ) [inline csunplugged/utils/errors/EmptyMarkdownFileError.py]: Should be `Returns:` and 4 spaces indentation on next line.

File: csunplugged/utils/errors/InvalidConfigFileError.py [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+    def __init__(self, yaml_file_path):
+        """Create error for invalid config file."""
+        super().__init__()
+        self.yaml_file_path = yaml_file_path
+
+    def __str__(self):
+        """
+        Override default error string.
+
+        Return:
```
Comment (JackMorganNZ) [inline csunplugged/utils/errors/InvalidConfigFileError.py]: Should be `Returns:` and 4 spaces indentation on next line.

File: csunplugged/utils/errors/KeyNotFoundError.py [hunk 16]
```diff
@@ -0,0 +1,34 @@
...
+        super().__init__()
+        self.config_file_path = config_file_path
+        self.key = key
+        self.field = field
+
+    def __str__(self):
+        """
+        Override default error string.
+
+        Return:
```
Comment (JackMorganNZ) [inline csunplugged/utils/errors/KeyNotFoundError.py]: Should be `Returns:` and 4 spaces indentation on next line.

File: csunplugged/utils/errors/MissingRequiredFieldError.py [hunk 17]
```diff
@@ -0,0 +1,38 @@
...
+        super().__init__()
+        self.config_file_path = config_file_path
+        self.required_fields = required_fields
+        self.model = model
+
+    def __str__(self):
+        """
+        Override default error string.
+
+        Return:
```
Comment (JackMorganNZ) [inline csunplugged/utils/errors/MissingRequiredFieldError.py]: Should be `Returns:`

File: csunplugged/utils/errors/NoHeadingFoundInMarkdownFileError.py [hunk 18]
```diff
@@ -0,0 +1,23 @@
...
+    def __init__(self, md_file_path):
+        """Create error for missing heading."""
+        super().__init__()
+        self.md_file_path = md_file_path
+
+    def __str__(self):
+        """
+        Override default error string.
+
+        Return:
```
Comment (JackMorganNZ) [inline csunplugged/utils/errors/NoHeadingFoundInMarkdownFileError.py]: Should be `Returns:`

File: csunplugged/utils/errors/InvalidConfigFileError.py [hunk 19]
```diff
@@ -0,0 +1,30 @@
...
+    """custom error for invalid config file."""
+
+    def __init__(self, yaml_file_path):
+        """Create error for invalid config file."""
+        super().__init__()
+        self.yaml_file_path = yaml_file_path
+
+    def __str__(self):
+        """
+        Override default error string.
```
Comment (JackMorganNZ) [inline csunplugged/utils/errors/InvalidConfigFileError.py]: I would have expected this line to be right after the `"""` but this seems to be valid syntax, should we try and always do it one way?

File: csunplugged/utils/check_required_files.py [hunk 20]
```diff
@@ -23,25 +23,25 @@ def render_scratch_images(scratch_images):
...
+            with open(filepath, "w") as scratch_temp_file:
                 scratch_temp_file.write(scratch_image.text)
 
 
-def find_image_files(images):
+def find_image_files(images, md_file_path):
     """Confirm each image is in static folder.
 
     Args:
         images (set): image file names
```
Comment (JackMorganNZ) [inline csunplugged/utils/check_required_files.py]: Missing argument in docstring.

File: docs/source/developer/django_setup.rst [hunk 21]
```diff
@@ -42,4 +42,22 @@ applications within the database.
...
 applications and other components of the CS Unplugged project.
 
+Loaders
+==============================================================================
+
+To populate the database with content we have written a series of custom loaders.
+The loaders for an application are found in the ``management/commands/`` directory, and
+there is approximately one loader for each configuration file.
+
+Besides just populate fields in the database, a loader is also responsible for checking
```
Comment (JackMorganNZ) [inline docs/source/developer/django_setup.rst]: Start of sentence doesn't really make sense, should be 'Besides populating fields...'

File: docs/source/developer/django_setup.rst [hunk 22]
```diff
@@ -42,4 +42,22 @@ applications within the database.
...
+that it's corresponding config file contains all the required fields, markdown files are
```
Comment (JackMorganNZ) [inline docs/source/developer/django_setup.rst]: "its" instead of "it's"

Comment (JackMorganNZ) [inline docs/source/developer/django_setup.rst]: "Markdown" instead of "markdown"

File: docs/source/developer/django_setup.rst [hunk 23]
```diff
@@ -42,4 +42,22 @@ applications within the database.
...
+not empty, and icons can be found. If any of these conditions are not met, then an error
+is thrown.
+
+Errors are defined in ``utils/errors/`` and should aim to be as descriptive and useful
+as possible as they will most often be read by an author and not necessarily a python
```
Comment (JackMorganNZ) [inline docs/source/developer/django_setup.rst]: "Python" instead of "python"

File: docs/source/developer/django_setup.rst [hunk 24]
```diff
@@ -42,4 +42,22 @@ applications within the database.
...
+developer.
+
 .. _Two Scoops of Django - Best Practices for Django 1.8: https://www.twoscoopspress.com/products/two-scoops-of-django-1-8
+
+
```
Comment (JackMorganNZ) [inline docs/source/developer/django_setup.rst]: Extra newline

File: docs/source/getting_started/installation.rst [hunk 25]
```diff
@@ -252,11 +252,6 @@ you're all set to contribute to the CS Unplugged project.
 .. _SourceTree: https://www.sourcetreeapp.com/
 .. _GitHub website: https://github.com/
 .. _SSH keys: https://help.github.com/articles/connecting-to-github-with-ssh/
-.. _Setting your username in Git: https://help.github.com/articles/setting-your-username-in-git/
-.. _Setting your email in Git: https://help.github.com/articles/setting-your-email-in-git/
-.. _keep your email address private on GitHub: https://help.github.com/articles/keeping-your-email-address-private/
-.. _Docker: https://www.docker.com/
-.. _Docker Store: https://store.docker.com/search?type=edition&offering=community
```
Comment (JackMorganNZ) [inline docs/source/getting_started/installation.rst]: Why are these deleted? This has broken a lot of links in the installation page.

File: csunplugged/topics/content/en/curriculum-areas.yaml [hunk 26]
```diff
@@ -1,16 +1,72 @@
...
 
 art:
   name: Art
 
-design:
-  name: Design
+science:
+  name: Science
+  children:
+    biology:
```
Comment (hayleyavw) [inline csunplugged/topics/content/en/curriculum-areas.yaml]: it can be whatever you like. I removed the ``science-`` prefix since it's just a duplicate of the parent.

File: csunplugged/topics/management/commands/_CurriculumAreasLoader.py [hunk 27]
```diff
@@ -1,47 +1,101 @@
...
 
 class CurriculumAreasLoader(BaseLoader):
     """Loader for curriculum area content."""
 
-    def __init__(self, curriculum_areas_file, BASE_PATH):
+    def __init__(self, structure_file_path, BASE_PATH):
         """Create the loader for loading curriculum areas.
 
         Args:
             curriculum_areas_file: File path to YAML file (string)
```
Comment (hayleyavw) [inline csunplugged/topics/management/commands/_CurriculumAreasLoader.py]: I see your point, but the new file name is more consistent across loaders and is more descriptive as it tells you it is a file *path*, not a loaded file. After working on the loaders for 2 weeks straight, I found this much easier to understand.

File: csunplugged/utils/BaseLoader.py (hunk 4 above)
Comment (hayleyavw) [inline csunplugged/utils/BaseLoader.py]: good catch. This was here because at the time of writing video was not a default processor, this has since been updated so I have removed this step.

File: docs/source/author/topics.rst [hunk 28]
```diff
@@ -17,30 +18,25 @@ diagram.
...
-    - A programming exercise can have **language implementations**, which contain
-      language specific expected values, hints, and possible solutions.
-      For example: an exercise may have implementations available in Scratch and
-      Python.
+  - A **topic** can also contain **programming exercises**.
 
-- **Learning outcomes** and **curriculum areas** are defined at a language
-  level, so can be used by all topic content.
+    - A **programming exercise** can use different **programming languages** and be set at
+      different **difficulties**.
```
Comment (JackMorganNZ) [inline docs/source/author/topics.rst]: set at 'a specific' difficulty (since you are talking about a singular exercise

File: docs/source/author/topics.rst [hunk 29]
```diff
@@ -80,1007 +76,634 @@ the project's topic application, where:
...
-  We don't recommend using numbered slugs (for example: ``integration-1``) as
-  ordering may change but a slug should never change.
+- **Purpose:** Defines the learning outcomes avilable for all topics.
 
-  - For example:
+- **Required Fields:**
 
-    .. code-block:: yaml
+  - ``<key> : <value>`` Key value pairs. The key will be used in other configuration
+    files to reference this particluar learning objective. The value is the learning
```
Comment (JackMorganNZ) [inline docs/source/author/topics.rst]: 'outcome' instead of 'objective'

File: docs/source/author/topics.rst [hunk 30]
```diff
@@ -80,1007 +76,634 @@ the project's topic application, where:
...
+    objective text that will be displayed to the user).
```
Comment (JackMorganNZ) [inline docs/source/author/topics.rst]: 'outcome' instead of 'objective'

File: docs/source/author/topics.rst [hunk 31]
```diff
@@ -17,30 +18,25 @@ diagram.
...
 
-- **Learning outcomes** and **curriculum areas** are defined at a language
-  level, so can be used by all topic content.
+    - A **programming exercise** can use different **programming languages** and be set at
+      different **difficulties**.
 
-- Also defined at the language level is **languages** and **difficulties** for
-  programming exercises.
+- **Learning outcomes**, **curriculum areas**, **programming languages** and
+  **difficulties** are defined at a language level, so can be used by all topic content.
```
Comment (JackMorganNZ) [inline docs/source/author/topics.rst]: Suggestion: so can be used by all topic content 'of that language'.

File: docs/source/author/topics.rst [hunk 32]
```diff
@@ -80,1007 +76,634 @@ the project's topic application, where:
...
-attributes (for example: the difficulty of a programming exercise).
-If you haven't used the YAML configuration files before, see
-:doc:`understanding_configuration_files`.
+.. The following image can copied for be edits here:
+.. image:: ../_static/img/topics_adding_learning_outcomes_flowchart.png
 
-The diagram below shows an example of YAML file locations for the
-``content/en/`` language directory, where:
+You will now be able to add learning outcomes to lessons and programming
+exercises by referencing the keys you specified in the learning outcomes config
```
Comment (JackMorganNZ) [inline docs/source/author/topics.rst]: 'configuration' instead of 'config'

File: docs/source/author/topics.rst [hunk 33]
```diff
@@ -80,1007 +76,634 @@ the project's topic application, where:
...
-  - For example:
-
-    .. code-block:: yaml
+.. The following image can copied for be edits here:
+.. image:: ../_static/img/topics_adding_curriculum_areas_flowchart.png
 
-      learning-outcomes: learning-outcomes.yaml
+You will now be able to add curriculum areas to lessons and curriculum
+integrations by referencing the keys you specified in the curriculum areas
+config file.
```
Comment (JackMorganNZ) [inline docs/source/author/topics.rst]: 'configuration' instead of 'config'

File: docs/source/author/topics.rst [hunk 34]
```diff
@@ -80,1007 +76,634 @@ the project's topic application, where:
...
-This file defines the attributes of a specific topic, including connected
-unit plan, programming exercise, and curriculum integration configuration
-files.
+.. The following image can copied for be edits here:
+.. image:: ../_static/img/topics_adding_programming_exercises_flowchart.png
 
-The location of this file is within a topic directory.
-This file is listed in the :ref:`application-structure-file`.
+You will now be able to add programming exercises to lessons by referencing the
+keys you specified in the programming exercises config file.
```
Comment (JackMorganNZ) [inline docs/source/author/topics.rst]: 'configuration' instead of 'config'

File: docs/source/developer/django_setup.rst [hunk 35]
```diff
@@ -42,4 +42,20 @@ applications within the database.
...
 
+Loaders
+==============================================================================
+
+To populate the database with content we have written a series of custom loaders.
+The loaders for an application are found in the ``management/commands/`` directory, and
+there is approximately one loader for each configuration file.
+
+Besides populating fields in the database, a loader is also responsible for checking
+that its corresponding config file contains all the required fields, Markdown files are
```
Comment (JackMorganNZ) [inline docs/source/developer/django_setup.rst]: 'configuration' instead of 'config'

File: docs/source/author/topics.rst [hunk 36]
```diff
@@ -80,1007 +76,634 @@ the project's topic application, where:
...
 
-This file must contain the following key/value pairs:
-
-- ``slug`` - The URL slug for this topic (see :ref:`what-is-a-slug`).
-
-  - For example:
+.. note::
 
-    .. code-block:: yaml
+  If the exercise includes learning outcomes, then the corresponding config
```
Comment (JackMorganNZ) [inline docs/source/author/topics.rst]: 'configuration' instead of 'config'

File: docs/source/author/topics.rst [hunk 37]
```diff
@@ -80,1007 +76,634 @@ the project's topic application, where:
...
+  file will also need to be added.
```
Comment (JackMorganNZ) [inline docs/source/author/topics.rst]: will also need to be added 'or updated to include new learning outcomes.'

File: docs/source/author/topics.rst [hunk 38]
```diff
@@ -80,1007 +76,634 @@ the project's topic application, where:
...
+.. note::
 
-  - The following slugs are not allowed for topics:
+  A programming exercise is split into four parts, these are:
 
-    - ``curriculum-integrations``
+    1. The exercise description
+    2. The expected output
+    3. Hints (optional)
+    4. An example solution
```
Comment (JackMorganNZ) [inline docs/source/author/topics.rst]: '4. Example solution/s' instead of '4. An example solution'

File: docs/source/author/topics.rst [hunk 39]
```diff
@@ -80,1007 +76,634 @@ the project's topic application, where:
...
+.. The following image can copied for be edits here:
+.. image:: ../_static/img/topics_adding_curriculum_integrations_flowchart.png
 
-    .. code-block:: yaml
+.. note ::
 
-      unit-plans:
-        - unit-plan/unit-plan.yaml
+  If the integration includes curriculum areas and/or prerequisite lessons,
+  then the corresponding config and content files will also need to be added.
```
Comment (JackMorganNZ) [inline docs/source/author/topics.rst]: 'configuration' instead of 'config'

File: docs/source/author/topics.rst [hunk 40]
```diff
@@ -80,1007 +76,634 @@ the project's topic application, where:
...
+    range from learning outcome text, to the structure and attributes of a lesson).
+    There are two things to remember with keys:
 
-    .. code-block:: yaml
+    - They must be written in lowercase and use hyphens instead of spaces.
 
-      other-resources-md-file: other-resources.md
+    - They must be exact matches to work, for example, if you name a lesson
+      ``bits-and-bytes``, referencing it in another configuration file as
+      ``bytes-and-bits`` will raise an error.
```
Comment (JackMorganNZ) [inline docs/source/author/topics.rst]: Great example! :+1:

File: docs/source/author/topics.rst [hunk 41]
```diff
@@ -80,1007 +76,634 @@ the project's topic application, where:
...
 
-A complete topic file may look like the following:
+.. note::
 
-.. code-block:: yaml
+  - Some of the keys have angle brackets around them, ``<like so>``. This means that they
+    are variables and you can call them whatever you like in your configuration file
+    (without the angle brackets).
```
Comment (JackMorganNZ) [inline docs/source/author/topics.rst]: Will see how easy this is to understand for authors, hopefully should be fine.

File: docs/source/author/topics.rst [hunk 42]
```diff
@@ -80,1007 +76,634 @@ the project's topic application, where:
...
 
-This file defines the attributes of a unit plan, including all lessons (and
-their respective attributes) for the unit plan.
-
-The location of this file is within a unit plan directory.
-These files are listed in a :ref:`topic-file`.
+- **File Name:** ``structure.yaml``
 
-This file must contain the following key/value pairs:
+- **Location:** ``topics/content/<language>/``
```
Comment (JackMorganNZ) [inline docs/source/author/topics.rst]: Same angle bracket syntax as mentioned above? Would it be worth using a different bracket type?

File: docs/source/author/topics.rst [hunk 43]
```diff
@@ -80,1007 +76,634 @@ the project's topic application, where:
...
 
-          curriculum-areas:
-            - maths
+        - ``max-age:`` The suggested maximum age group to teach this lesson to.
 
-    - ``resources-classroom`` - A list of Markdown text of classroom resources
-      required for this lesson resources-classroom.
+        - ``number:`` The number order for this lesson.
+          Lessons are sorted by minimum age, maximum age, then number so lessons in
+          different age ranges can use the same number without conflict.
```
Comment (JackMorganNZ) [inline docs/source/author/topics.rst]: Might be better as "Lessons are group by their minimum age and maximum age, then ordered by number so lessons in different age groups can use the same number without conflict."

File: docs/source/author/topics.rst [hunk 44]
```diff
@@ -80,1007 +76,634 @@ the project's topic application, where:
...
+      - ``name:`` The name of the curriculum area (this is what will be displayed to the
+        user).
 
-        .. code-block:: yaml
+    - **Optional Fields:**
 
-          prerequisite-lessons:
-            - unit-plan/introduction-to-bits
-            - unit-plan/counting-bits
+      - ``children:`` A list of sub-curriculm areas (see example file below). Each child
```
Comment (JackMorganNZ) [inline docs/source/author/topics.rst]: 'curriculum' instead of 'curriculm'

File: docs/source/author/topics.rst [hunk 45]
```diff
@@ -80,1007 +76,634 @@ the project's topic application, where:
...
+.. note::
 
-      lessons:
-        python:
-          ...language data here...
-        scratch:
-          ...language data here...
+  When including a curriculum area in another configuration file, adding a child
+  curriculum area will automatically add the parent curriculum area, you do not need to
+  specify this manually. For example, adding "geometry" means that "maths" is
```
Comment (JackMorganNZ) [inline docs/source/author/topics.rst]: Would it be better to have `geometry` and `maths` in code blocks, as we are referencing keys?

File: docs/source/author/topics.rst [hunk 46]
```diff
@@ -80,1007 +76,634 @@ the project's topic application, where:
...
 
-      - Each language slug must contain the following values:
+- **Referenced In:** ``topics/content/<language>/<topic-name>/<topic-name>.yaml``
 
-        - ``expected-result`` - The Markdown file containing the expected
-          result for the language implementation of the exercise.
-          This is essentially the text for the expected result section for
-          this language on the website for this exercise.
-          The file path is relative to the YAML file.
+- **Purpose:** This file defines the programming exercises (their respective attributes)
```
Comment (JackMorganNZ) [inline docs/source/author/topics.rst]: 'and their respective attributes' instead of 'their respective attributes'

File: docs/source/author/topics.rst [hunk 47]
```diff
@@ -80,1007 +76,634 @@ the project's topic application, where:
...
+      - ``curriculum-areas:`` A list of keys corresponding to other curriculum areas
+        that this activity could be used in.
 
-- **Curriculum area slugs** - A slug listed for each curriculum area
-  (see :ref:`what-is-a-slug`).
+    - **Optional Fields:**
 
-  - For example:
+      - ``prerequisite-lessons:`` A list of unit plan keys containing lessons that are
+        expected to be completed before attemping this activity.
```
Comment (JackMorganNZ) [inline docs/source/author/topics.rst]: 'attempting' instead of 'attemping'

File: csunplugged/topics/content/en/curriculum-areas.yaml (hunk 26 above)
Comment (JackMorganNZ) [inline csunplugged/topics/content/en/curriculum-areas.yaml]: :+1:

File: csunplugged/resources/views/sorting_network.py (hunk 2 above)
Comment (JackMorganNZ) [inline csunplugged/resources/views/sorting_network.py]: We have decided to stick with double quotes.

File: docs/source/author/topics.rst [hunk 48]
```diff
@@ -80,1007 +77,622 @@ the project's topic application, where:
...
+You will now be able to add programming exercises to lessons by referencing the
+keys you specified in the programming exercises configuration file.
 
-    .. code-block:: yaml
-
-      slug: binary-numbers
+.. note::
 
-  - The following slugs are not allowed for topics:
+  If the exercise includes learning outcomes, then the corresponding config
```
Comment (JackMorganNZ) [inline docs/source/author/topics.rst]: 'configuration' instead of 'config'
